lib/tasks/npm-task.js

```diff
@@ -63,7 +63,11 @@ class NpmTask extends Task {
 
       if (semver.gte(version, '2.0.0')) {
         logger.warn('yarn --version: %s', version);
-        this.ui.writeWarnLine(`Yarn v2 is not fully supported. Proceeding with yarn: ${version}`);
+        let yarnConfig = await this.yarn(['config', 'get', 'nodeLinker']);
+        let nodeLinker = yarnConfig.stdout.trim();
+        if (nodeLinker !== 'node-modules') {
+          this.ui.writeWarnLine(`Yarn v2 is not fully supported. Proceeding with yarn: ${version}`);
+        }
```

Comment:
Does `yarn config get nodeLinker` work on all Yarns (e.g. 1.x, 2.x, 3.x)? If it does, maybe we could just do that instead of calling `yarn --version` **and** `yarn config get nodeLinker`?

Reply:
Locally in a yarn@1 repo, it does seem like it works (just returns `"undefined"`)

```
❯ yarn config get nodeLinker
warning package.json: No license field
undefined
```

Reply:
Im not opposed to doing that since the `yarn config get`command is present in all versions of Yarn, however, how would we distinguish the difference between yarn 1 vs yarn 3 , specifically for the warning message? `nodeLinker === undefined` as Yarn 1 whereas `[pnpm,pnp,node_modules].includes(nodeLinker)` for Yarn 3?  Additionally, I'm not sure if there is a case for Yarn 3 where the nodeLinker is undefined.

[Reference for Yarn 3 available values](https://yarnpkg.com/configuration/yarnrc#nodeLinker)

Reply:
Maybe I'm misunderstanding the output on yarn 3, but won't `yarn config get nodeLinker` always reply consistently?

- Yarn 1 - `undefined`
- Yarn 2 - `pnp`
- Yarn 3 - whichever linker you have setup

I would imagine the logic to be to explicitly check for `pnp` (and trigger the warning) or we could probably also make it so that If we see any value _other than_ `undefined` or `node-modules` we would trigger the warning.

Does that make sense? Would that work?

Reply:
Ah okay. I see what you are saying. From thinking about it, i think that would work fine. I'll push out another revision to account for it. Thanks @rwjblue!

Reply:
Upon further investigation, I think we should keep the version call (although I would like to remove that extra call) . If we remove it, it would:

1) provide an inconsistent API for the `packageManager`. (AKA checkNPM and checkPNPM return name & version whereas yarn would return only the name)

2) Refactoring the `npm-tasks.js` to where version is not needed, would cause some issues for people on older versions of NPM. Basically, other PMs need the version number as well.

Reply:
Sounds good, thanks for digging!